crates/reporting/src/error/type.rs

```diff
@@ -1237,8 +1268,8 @@ fn to_expr_report<'b>(
                 &category,
                 found,
                 expected_type,
-                region,
-                Some(expr_region),
```

Comment:
These were previously swapped, meaning we wouldn't show an expression and some context, but just the expression. Errors should be more useful (a.k.a. have more context) now.